crates/wash-lib/src/build.rs

```diff
@@ -315,15 +316,25 @@ fn build_tinygo_actor(
     }
 
     // Generate wit-bindgen code for Golang actors which are components-first
-    if let WasmTarget::WasiPreview1 | WasmTarget::WasiPreview2 = &actor_config.wasm_target {
+    //
+    // Running wit-bindgen via go generate is only for WIT-enabled projects, so we must limit
+    // to only projects that have their WIT defined in the expected top level wit directory
+    //
+    // While wasmcloud and it's tooling is WIT-first, it is possible to build preview1/preview2
+    // components that are *not* WIT enabled. To determine whether the project is WIT-enabled
+    // we check for the `wit` directory which would be passed through to bindgen.
+    let wit_dir = common_config.path.join("wit");
+    if let (WasmTarget::WasiPreview1 | WasmTarget::WasiPreview2, true) =
```

Comment:
We should probably print out a warning if they have the wasm target set but the wit is _not_ there. It would help users that put that in, then find out they aren't building a component

Reply:
The problem is that it's possible to build a component that isn't WIT-based, it's just that we expect components be WIT based right now. 

Rather than do this, what I'll do is key off of `wit_world`.

Reply:
To address the possibility of the wit dir being missing I'll ad a specific error message for that (can't do a warning, since this is `wash-lib`)